tests/test_init.py

```diff
@@ -19,90 +19,241 @@
-                      (qml.init.cvqnn_layers_phi_uniform, rpt_uni, (repeat, n_if)),
-                      (qml.init.cvqnn_layers_varphi_uniform, rpt_uni, (repeat, n_wires)),
-                      (qml.init.cvqnn_layers_r_uniform, rpt_uni, (repeat, n_wires)),
-                      (qml.init.cvqnn_layers_phi_r_uniform, rpt_uni, (repeat, n_wires)),
-                      (qml.init.cvqnn_layers_a_uniform, rpt_uni, (repeat, n_wires)),
-                      (qml.init.cvqnn_layers_phi_a_uniform, rpt_uni, (repeat, n_wires)),
-                      (qml.init.cvqnn_layers_kappa_uniform, rpt_uni, (repeat, n_wires)),
-                      (qml.init.interferometer_theta_uniform, base_uni, (n_if,)),
-                      (qml.init.interferometer_phi_uniform, base_uni, (n_if,)),
-                      (qml.init.interferometer_varphi_uniform, base_uni, (n_wires,)),
-                      (qml.init.qaoa_embedding_normal, rpt_nrml, (repeat, 2*n_wires)),
-                      (qml.init.qaoa_embedding_uniform, rpt_uni, (repeat, 2*n_wires)),
-                      (qml.init.qaoa_embedding_uniform, {'n_layers': 2, 'n_wires': 1, 'low': 0, 'high': 1}, (2, 1)),
-                      (qml.init.qaoa_embedding_uniform, {'n_layers': 2, 'n_wires': 2, 'low': 0, 'high': 1}, (2, 3)),
-                      (qml.init.qaoa_embedding_normal, {'n_layers': 2, 'n_wires': 1, 'mean': 0, 'std': 1}, (2, 1)),
-                      (qml.init.qaoa_embedding_normal, {'n_layers': 2, 'n_wires': 2, 'mean': 0, 'std': 1}, (2, 3)),
-                      (qml.init.simplified_two_design_initial_layer_uniform, {'n_wires': 1, 'low': 0, 'high': 1}, (1,)),
-                      (qml.init.simplified_two_design_initial_layer_uniform, {'n_wires': 3, 'low': 0, 'high': 1}, (3,)),
-                      (qml.init.simplified_two_design_initial_layer_normal, {'n_wires': 1, 'mean': 0, 'std': 1}, (1,)),
-                      (qml.init.simplified_two_design_initial_layer_normal, {'n_wires': 3, 'mean': 0, 'std': 1}, (3,)),
-                      (qml.init.simplified_two_design_weights_uniform, {'n_layers': 2, 'n_wires': 1, 'low': 0, 'high': 1}, (0,)),
-                      (qml.init.simplified_two_design_weights_uniform, {'n_layers': 2, 'n_wires': 2, 'low': 0, 'high': 1}, (2, 1, 2)),
-                      (qml.init.simplified_two_design_weights_uniform, {'n_layers': 2, 'n_wires': 4, 'low': 0, 'high': 1}, (2, 3, 2)),
-                      (qml.init.simplified_two_design_weights_normal, {'n_layers': 2, 'n_wires': 1, 'mean': 0, 'std': 1}, (0, )),
-                      (qml.init.simplified_two_design_weights_normal, {'n_layers': 2, 'n_wires': 2, 'mean': 0, 'std': 1}, (2, 1, 2)),
-                      (qml.init.simplified_two_design_weights_normal, {'n_layers': 2, 'n_wires': 4, 'mean': 0, 'std': 1}, (2, 3, 2)),
-                      (qml.init.basic_entangler_layers_normal, {'n_layers': 2, 'n_wires': 1, 'mean': 0, 'std': 1},
+INIT_KWARGS_SHAPES = [(qml.init.random_layers_normal,
```

Comment:
There should now be ``wires=1`` cases for every init function.

Reply:
My suggestion for this list in the short term would be to split it up based on the function that is being called:
```python
RANDOM_NORMAL = [
                       (qml.init.random_layers_normal,
                       {'n_layers': 2, 'n_wires': 3, 'n_rots': 10, 'mean': 0, 'std': 1},
                       (2, 10)),
                      (qml.init.random_layers_normal,
                       {'n_layers': 2, 'n_wires': 1, 'n_rots': 10, 'mean': 0, 'std': 1},
                       (2, 10)),
                      (qml.init.random_layers_normal,
                       {'n_layers': 2, 'n_wires': 3, 'n_rots': None, 'mean': 0, 'std': 1},
                       (2, 3))
                       ]

# All the other types defined in a similar fashion

# Grouping terms within the base group
INIT_KWARGS_SHAPES = [
                       *RANDOM_NORMAL, *RANDOM_UNIFORM,
                       *STRONG_ENT_NORMAL, *STRONG_ENT_UNIFORM,
                       (....)
                       ]
```
This will ensure that there is a clear logical separation between cases and improve readability. It does not tackle the length of these lists but makes it easier to have an overview.

Reply:
You are right, but since this test file is likely to disappear in future, I want to avoid doing any cosmetic improvements at this stage :) 

The edge case tests however were important for functionality - and unearthed indeed one or two problems.

Reply:
Sure!